File: code_combat/7_Sarven_Desert/286-Weakest_Quickest/weekest_quickest.py
def findWeakestEnemy():
    enemies = hero.findEnemies()
    weakest = None
    leastHealth = 99999
    enemyIndex = 0
    while enemyIndex < len(enemies):
        enemy = enemies[enemyIndex]
        if enemy.health < leastHealth: 
            weakest = enemy
            leastHealth = enemy.health
        enemyIndex += 1

    return weakest


while True:
    weakestShaman = findWeakestEnemy()
    if weakestShaman:
        hero.attack(weakestShaman)

File: projects/code_combat/7_Sarven_Desert/286-Weakest_Quickest/weekestQuickest.js
// Defeat shamans to survive.

// This function finds the weakest enemy.
function findWeakestEnemy() {
    var enemies = hero.findEnemies();
    var weakest = null;
    var leastHealth = 99999;
    var enemyIndex = 0;
    // Loop through enemies:
    while (enemyIndex < enemies.length) {
        var enemy = enemies[enemyIndex];
        // If an enemy's health is less than leastHealth:
        if (enemy.health < leastHealth) {
            // Make it the leastHealth 
            weakest = enemy;
            // and set leastHealth to its health.
            leastHealth = enemy.health;
        }
        enemyIndex++;
    }
    return weakest;
}

while(true) {
    // Find the weakest enemy with the function:
    var weakestShaman = findWeakestEnemy();
    // If the weakest enemy exists:
    if (weakestShaman) {
        // Attack it!
        hero.attack(weakestShaman);
    }
}
